CAM2ImageArchiver/CameraHandler.py

```diff
@@ -92,7 +96,19 @@ def run(self):
                             cam_directory, camera.id,
                             datetime.datetime.fromtimestamp(
                                 frame_timestamp).strftime('%Y-%m-%d_%H-%M-%S-%f'))
-                        cv2.imwrite(file_name, frame)
+                        
+                        if self.image_difference_percentage:
+                            if type(camera.last_frame) == type(None) or abs(np.sum(camera.last_frame - frame))/abs(np.sum(camera.last_frame)) * 100 >= self.image_difference_percentage:
```

Comment:
I think the avg value is calculated incorrectly. This is a better calculation:
`if frame.size != 0 and (type(camera.last_frame) == type(None) or (np.count_nonzero(cv2.absdiff(camera.last_frame, frame)) * 100)/ frame.size >= self.image_difference_percentag)`
This should give you the percentage of pix that changed between the two frames. I beleave this is correct although you should test it.